### PR TITLE
Sprite Sheet abstraction

### PR DESCRIPTION
**Pull request by Burke-Daniel**
This is super basic (can only really use Sprite Sheets where every sprite is the same size), but should be enough for now and we can add to it down the road (@colton-smith if you think of anything that would make your life easier for the top-down RPG lmk and I can add in).

### CODE REVIEW

File: Aquarius/Src/Aquarius/Renderer/QuadTexCoords.h [hunk 1]
```diff
@@ -11,6 +11,12 @@ namespace Aquarius {
 	class QuadTexCoords
 	{
 	public: 
+		QuadTexCoords()
+			: m_data(std::array<glm::vec2, 4>{ glm::vec2{ 0, 0 },
+											   glm::vec2{ 0, 0 },
+											   glm::vec2{ 0, 0 },
+											   glm::vec2{ 0, 0 } })
+		{}
```
Comment (colton-smith) [inline Aquarius/Src/Aquarius/Renderer/QuadTexCoords.h]: Must be visual studio tab problem again

File: Aquarius/Src/Aquarius/Renderer/SpriteSheet.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		m_Texture = texture;
+		m_TotalWidth = texture->getWidth();
+		m_TotalHeight = texture->getHeight();
+		m_SpriteWidth = spriteWidth;
+		m_SpriteHeight = spriteHeight;
+	}
+
+	QuadTexCoords SpriteSheet::getSpriteCoords(uint32_t i, uint32_t j) const
+	{
+		// Bottom Left Zero Indexed
```
Comment (colton-smith) [inline Aquarius/Src/Aquarius/Renderer/SpriteSheet.cpp:25]: So the bottom left of the sprite sheet is (0,0)?

File: Sandbox/Sandbox.cpp [hunk 3]
```diff
@@ -43,6 +43,9 @@ void SandboxLayer::onCreation()
     m_texture = std::make_shared<Aquarius::Texture>("Sandbox/Assets/greenguy.png", texConfig, true);
     m_texture->bind(0);
 
+    Aquarius::SpriteSheet sheet(m_texture.get(), 32, 32);
+    m_uv = sheet.getSpriteCoords(0, 1);
```
Comment (colton-smith) [inline Sandbox/Sandbox.cpp:47]: So much nicer than manual lmfao

File: Aquarius/Src/Aquarius/Renderer/SpriteSheet.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "Aquarius/Renderer/Texture.h"
```
Comment (colton-smith) [inline Aquarius/Src/Aquarius/Renderer/SpriteSheet.h]: Is this include necessary since you forward declare?

**Review by colton-smith — APPROVED** [review]
Some odd spacing in QuadTexCoords.h and a possible unnecessary include then you're g2g

File: Aquarius/Src/Aquarius/Renderer/QuadTexCoords.h (hunk 1 above)
Comment (Burke-Daniel) [reply Aquarius/Src/Aquarius/Renderer/QuadTexCoords.h]: Goddam lol

File: Aquarius/Src/Aquarius/Renderer/SpriteSheet.cpp (hunk 2 above)
Comment (Burke-Daniel) [reply Aquarius/Src/Aquarius/Renderer/SpriteSheet.cpp:25]: Correct

File: Sandbox/Sandbox.cpp (hunk 3 above)
Comment (Burke-Daniel) [reply Sandbox/Sandbox.cpp:47]: Hahah yeah for sure

File: Aquarius/Src/Aquarius/Renderer/SpriteSheet.h (hunk 4 above)
Comment (Burke-Daniel) [reply Aquarius/Src/Aquarius/Renderer/SpriteSheet.h]: Nope, will remove

File: Aquarius/Src/Aquarius/Renderer/SpriteSheet.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <vector>
+
```
Comment (liamjwarren) [inline Aquarius/Src/Aquarius/Renderer/SpriteSheet.h:6]: Add line here

**Review by liamjwarren — APPROVED** [review]
Sweet, one style comment but otherwise GTM

**Pull request merged**